unlock-app/src/pages/staticdemo.js

```diff
@@ -95,18 +99,32 @@ const Demo = ({ lock, locks }) => {
 Demo.propTypes = {
   lock: PropTypes.string,
   locks: UnlockPropTypes.locks.isRequired,
+  hasKey: PropTypes.bool.isRequired,
 }
 
 Demo.defaultProps = {
   lock: '',
 }
 
-export const mapStateToProps = ({ locks, router }) => {
+export const mapStateToProps = ({ locks, keys, router }) => {
   const match = router.location.pathname.match(LOCK_PATH_NAME_REGEXP)
   const lock = match ? match[1] : null
+  let hasKey = false
+  const thislock = locks[lock]
+  if (thislock) {
+    for (let k of Object.values(keys)) {
```

Comment:
So this logic is the one in `showUnless...`, right? I think we should extract it in a PR before this one?